compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/repositories/RemotePackageRepository.java

```diff
@@ -133,6 +133,8 @@ public static RemotePackageRepository from(Environment environment, Path cacheDi
             }
         } catch (ConnectionErrorException e) {
             // ignore connect to remote repo failure
+            // set `offline` true
+            this.isOffline = true;
```

Comment:
Doing this might take the repo indefinitely offline. This might be an issue when using project api in a service. Can we make the offline only affect the current compilation

Reply:
IMO what we need to do is simply add a diagnostic and continue. But we would need to implement diagnostic reporting in Package Resolution phase for that.

Reply:
Shall we just print the error as a warning and continue the run?? @hevayo wdyt??